### ISSUE TITLE
[TASK] Verify os.after_install_chroot_commands and after-upgrade-chroot cloud-init config

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#5567]
## Verify `os.after_install_chroot_commands` works
- Install single node Harvester v1.2.1, please verify two paths:
    - PXE way
    - Interactive way (use remote config: https://docs.harvesterhci.io/assets/images/import-ssh-keys-9171ecc6f8d996859183a4147e10044f.png)
- Use the `os.after_install_chroot_commands` configuration to run a command to create a file in OS filesystem (https://docs.harvesterhci.io/v1.2/install/harvester-configuration#osafter_install_chroot_commands)
    - for example:
        ```
        os:
          after_install_chroot_commands:
            - echo install > /test-file-install
        ```
- After the installation succeeds, check the file present in the system.

## Verify `after-upgrade-chroot `
- Leverage `after-upgrade-chroot` stage to create another file in the OS filesystem after an upgrade. (https://rancher.github.io/elemental-toolkit/docs/customizing/runtime_persistent_changes/#example)
    - you can create a file `/oem/91_upgrade.yaml`
    - The stage should contain a command to create a file, like:
      ```
      stages:
      name: "Install something"
      stages:
         after-upgrade-chroot:
           - commands:
              - echo upgrade > /test-file-upgrade
      ```
- Upgrade Harvester to the same version or v1.2-head version. After the upgrade is done, check the file present in the system.

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.2-1a44ad32-head` (24/04/11) and `v1.2-f414942f-head` (24/04/08). 

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Verify the <strong>os.after_install_chroot_commands</strong>  works on the PXE installation $~~$</summary>

  1. After we set the `os.after_install_chroot_commands` configuration on the config-create.yml
  2. The **test-file-install** file exists after pxe installation
  ```
  harvester-node-0:/ # ls /test-file-install
  /test-file-install
  harvester-node-0:/ # cat /test-file-install
  install
  ```

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Verify the <strong>os.after_install_chroot_commands</strong> works on the interactive installation $~~$</summary>

  1. After we set the `os.after_install_chroot_commands` configuration in the `config` file
  2. And specify it to the URL path of the remote configuration file during installation
  3. The **test-file-install** file exists after interaction installation.
  ```
  harv121-upd:~ # ls /test-file-install
  /test-file-install
  harv121-upd:~ # cat /test-file-install
  install
  ```

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Verify the <strong>after-upgrade-chroot</strong> works after upgrade from v1.2.1 to v1.2-head $~~$</summary>

  1. Continue from the `os.after_install_chroot_commands`  scenario
  2. And set the `after-upgrade-chroot` in the `/oem/91_upgrade.yaml` file 
  3. The **test-file-install** file exists after upgrade from v1.2.1 to `v1.2-head`
  ```
  harvester-node-0:~ # ls /test-file-upgrade
  /test-file-upgrade
  harvester-node-0:~ # cat /test-file-upgrade
  upgrade
  ```
  ![image](https://github.com/harvester/harvester/assets/29251855/1abae660-6960-481e-abc5-4088b903237a)


 </details>

### Test Information
* Test Environment: Single nodes Harvester on local kvm machine
* Harvester version:  `v1.2-1a44ad32-head` (24/04/11) and `v1.2-f414942f-head` (24/04/08).

### Verify Steps
<details><summary>Check the <strong>os.after_install_chroot_commands</strong> works on the PXE installation</summary>

  1. Add the `os.after_install_chroot_commands` to the config-join.yml of the ipxe-example 
  ```
  os:
  after_install_chroot_commands:
    - echo install > /test-file-install
  ```
  2. Provision the single node Harvester 
  3. After installation complete, ssh to the node machine
  4. Check the `/test-file-install` file exists and the content is correct

</details>

<details><summary>Check the <strong>os.after_install_chroot_commands</strong> works on the interactive installation</summary>

  1. Prepare a config file with the `os.after_install_chroot_commands`
  ```
  os:
  after_install_chroot_commands:
    - echo install > /test-file-install
  ``` 
  2. Launch the http server to host the config file
  3. Provision the single node Harvester 
  4. Provide the remote URL of the Harvester configuration file during the interactive installation 
  5. Check the `/test-file-install` file exists and the content is correct

</details>


<details><summary>Check tthe <strong>after-upgrade-chroot</strong> works after upgrade from v1.2.1 to v1.2-head</summary>

  1. Continue from the previous `os.after_install_chroot_commands` scenario to prepare the v1.2.1 Harvester 
  2. ssh to the node machine
  3. Create the `/oem/91_upgrade.yaml` and add the following content 
  ```
  stages:
  name: "Install something"
  stages:
     after-upgrade-chroot:
       - commands:
          - echo upgrade > /test-file-upgrade
  ```
  4. Upgrade to v1.2-head
  5. Check the `test-file-upgrade` file exists and the content is correct

</details>

**Comment by bk201** [comment]
Thanks David!
